### PR TITLE
Mobile/bare react native patches

### PR DESCRIPTION
**Pull request by tanx**
These are the patches to the bare project template. **Do not merge, just for review**. Final PR will follow

### CODE REVIEW

File: mobile-bare/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+2. `lnd/mobile/build/android/Lndmobile.aar` -> `android/Lndmobile`
+
+## Running the app
+
+### Install the dependencies
+
+```
+npm install
+cd ios
+pod install
```
Comment (valentinewallace) [inline mobile-bare/README.md]: Ran into [this](https://github.com/CocoaPods/CocoaPods/issues/8138) problem. I was able to get it to work by downgrading to cocoapods 1.5.3, but the downgrade caused problems for `npm run ios` and the Xcode build.

File: mobile-bare/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+1. Open `ios/LightningApp.xcworkspace` from the finder on your mac
+2. From xcode set the active scheme to `Release`
+3. Set the build target as `Generic iOS device`
+4. Bump the version and build number in the project
+5. Build the Archive in the menu under `Product -> Archive`
+
+### Android
+
+1. Open Android Studio
```
Comment (valentinewallace) [inline mobile-bare/README.md]: TODO(val): practice release for iOS/Android :)

File: src/component/qrcode-scanner.js [hunk 3]
```diff
@@ -62,12 +67,19 @@ export default class QRCodeScanner extends React.Component {
...
         <View style={styles.wrapper}>
-          <BarCodeScanner
-            onBarCodeScanned={
+          <RNCamera
+            captureAudio={false}
+            androidCameraPermissionOptions={{
+              title: 'Permission to use camera',
+              message: 'Allow Lightning to use the camera',
+              buttonPositive: 'OK',
+              buttonNegative: 'Cancel',
```
Comment (valentinewallace) [inline src/component/qrcode-scanner.js]: Do we navigate back if the user presses "Cancel"?

File: mobile-bare/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+## Building lnd
+
+### Build dependencies
+
+* node 10 LTS
+* react-native-cli (npm install -g react-native-cli)
+* cocoapods 1.7.2
+* go 1.11.6
+* [gomobile](https://github.com/golang/go/wiki/Mobile)
+* [lnd mobile build tools](https://github.com/lightninglabs/lnd/tree/mobile-build-tool)
```
[the remark this answers is not in the document]
Comment (tanx) [reply mobile-bare/README.md]: Not sure what the status is on that. cc @Roasbeef @halseth

File: mobile-bare/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+==========
+
+## Getting started
+
+This project was bootstrapped using [Expo CLI as a bare project](https://blog.expo.io/you-can-now-use-expo-apis-in-any-react-native-app-7c3a93041331). You should only need `react-native-cli` for development not `expo-cli`. See the [React Native CLI Getting Started Guide](https://facebook.github.io/react-native/docs/getting-started.html)
+
+## Building lnd
+
+### Build dependencies
+
```
[the remark this answers is not in the document]
Comment (tanx) [reply mobile-bare/README.md]: Ok. I'll add those to the readme.

File: mobile-bare/README.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+* [lnd mobile build tools](https://github.com/lightninglabs/lnd/tree/mobile-build-tool)
+
+### Building
+
+```
+go get golang.org/x/mobile/cmd/gomobile
+cd lnd
+make clean
+gomobile init
+make mobile
```
[the remark this answers is not in the document]
Comment (tanx) [reply mobile-bare/README.md]: `make mobile` does perform both of them. But sure, we could.

File: mobile-bare/README.md (hunk 1 above)
Comment (tanx) [reply mobile-bare/README.md]: > I was able to get it to work by downgrading to cocoapods 1.5.3

As far as I understand that was necessary only for the Expo 32 build. The bare builds support the new cocoapods.

File: src/component/qrcode-scanner.js (hunk 3 above)
Comment (tanx) [reply src/component/qrcode-scanner.js]: That just declines the camera permission. No need to navigate back.

File: mobile-bare/README.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+Next copy the lnd mobile binaries to this project
+
+1. `lnd/mobile/build/ios/Lndmobile.framework` -> `ios/lightning`
+2. `lnd/mobile/build/android/Lndmobile.aar` -> `android/Lndmobile`
+
+## Running the app
+
+### Install the dependencies
+
```
[the remark this answers is not in the document]
Comment (valentinewallace) [reply mobile-bare/README.md]: I also had to `brew install watchman` but that might be a problem specific to my system..

File: mobile-bare/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (valentinewallace) [reply mobile-bare/README.md]: True true, might not be necessary.

File: mobile-bare/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tanx) [reply mobile-bare/README.md]: Ok adding watchman

[the remark this answers is not in the document]
Comment (tanx) [reply mobile-bare/README.md]: > Maybe this is too obvious, but I had to do `npm i` in the `lightning-app` directory as well to make sure the proto files were generated

Good point.

File: mobile-bare/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tanx) [reply mobile-bare/README.md]: Changed it

**Comment by tanx** [comment]
@valentinewallace thanks for the review. Closing this PR to open one the master after we've released our last `v0.1.x` build.

**Pull request closed without merging**